packages/lodestar-validator/src/slashingProtection/attestation/service.ts

```diff
@@ -0,0 +1,145 @@
+    if (sameTargetAtt) {
+      // Interchange format allows for attestations without signing_root, then assume root is equal
+      if (isEqualNonZeroRoot(this.config, sameTargetAtt.signingRoot, att.signingRoot)) {
+        return SafeStatus.SAME_DATA;
+      } else {
+        throw new InvalidAttestationError({code: InvalidAttestationErrorCode.DOUBLE_VOTE, att, prev: sameTargetAtt});
+      }
+    }
+
+    // Check for a surround vote
+    try {
+      await this.minMaxSurround.assertNoSurround(pubKey, {source: att.sourceEpoch, target: att.targetEpoch});
+    } catch (e) {
+      if (e instanceof SurroundAttestationError) {
+        const prev = await this.attestationByTarget.get(pubKey, e.type.att2Target).catch(() => null);
+        switch (e.type.code) {
+          case SurroundAttestationErrorCode.IS_SURROUNDING:
+            throw new InvalidAttestationError({code: InvalidAttestationErrorCode.NEW_SURROUNDS_PREV, att, prev});
+          case SurroundAttestationErrorCode.IS_SURROUNDED:
+            throw new InvalidAttestationError({code: InvalidAttestationErrorCode.PREV_SURROUNDS_NEW, att, prev});
+        }
+      }
+      throw e;
+    }
+
+    // Refuse to sign any attestation with:
+    // - source.epoch < min(att.source_epoch for att in data.signed_attestations if att.pubkey == attester_pubkey), OR
+    // - target_epoch <= min(att.target_epoch for att in data.signed_attestations if att.pubkey == attester_pubkey)
+    // (spec v4, Slashing Protection Database Interchange Format)
+    const attestationLowerBound = await this.attestationLowerBound.get(pubKey);
+    if (attestationLowerBound) {
+      const {minSourceEpoch, minTargetEpoch} = attestationLowerBound;
+      if (att.sourceEpoch < minSourceEpoch) {
+        throw new InvalidAttestationError({
+          code: InvalidAttestationErrorCode.SOURCE_LESS_THAN_LOWER_BOUND,
+          sourceEpoch: att.sourceEpoch,
+          minSourceEpoch,
+        });
+      }
+
+      if (att.targetEpoch <= minTargetEpoch) {
+        throw new InvalidAttestationError({
+          code: InvalidAttestationErrorCode.TARGET_LESS_THAN_OR_EQ_LOWER_BOUND,
+          targetEpoch: att.targetEpoch,
+          minTargetEpoch,
+        });
+      }
+    }
+
+    return SafeStatus.OK;
+  }
+
+  /**
+   * Insert an attestation into the slashing database
+   * This should *only* be called in the same (exclusive) transaction as `checkAttestation`
+   * so that the check isn't invalidated by a concurrent mutation
+   */
+  async insertAttestation(pubKey: BLSPubkey, att: SlashingProtectionAttestation): Promise<void> {
+    await this.attestationByTarget.set(pubKey, [att]);
+    await this.minMaxSurround.insertAttestation(pubKey, {source: att.sourceEpoch, target: att.targetEpoch});
+  }
+
+  /**
+   * Interchange import / export functionality
+   */
+  async importAttestations(pubkey: BLSPubkey, atts: SlashingProtectionAttestation[]): Promise<void> {
+    await this.attestationByTarget.set(pubkey, atts);
```

Comment:
ain't this gonna be set in insertAttestation?

Reply:
Yes, but this `importAttestations` a specialized method to import a (large) number of attestations from an interchange file. Could you developed your question more?

Reply:
Seems like you don't need this line as you are gonna overwrite those values with individual insertAttestation in code below.

Reply:
The line below calls `this.minMaxSurround.insertAttestation` which inserts to a different repository than `attestationByTarget`. Let me know if the naming is not clear